api/api.yaml

```
@@ -0,0 +1,481 @@
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewCard'
+      responses:
+        '201':
+          description: Card created
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Card'
+        '400':
+          description: Error decode json
+        '401':
+          description: Card not authorized
+        '500':
+          description: Internal error Card adding
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/user/card/list:
+    get:
+      summary: get all users card data 
+      description: list card credentials.
+      operationId: ListCards
+      responses:
+        '200':
+          description: Listed cards
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Card'
+  /api/user/gtext/add:
+    post:
+      summary: Add new text note
+      description: Add new text note
+      operationId: AddGtext
+      parameters:
+      requestBody:
+        description: Note add to the store
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewGtext'
+      responses:
+        '201':
+          description: Gtext created
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Gtext'
+        '400':
+          description: Error decode json
+        '401':
+          description: Gtext not authorized
+        '500':
+          description: Internal error Card adding
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/user/gtext/list:
+    get:
+      summary: get all users notes data 
+      description: list notes.
+      operationId: ListGtexts
+      responses:
+        '200':
+          description: Listed notes
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Gtext'
+        '204':
+          description: No content
+        '400':
+          description: Error decode json
+        '401':
+          description: User not authorized
+        '500':
+          description: Internal error site listing
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/user/file/add:
```

```suggestion
  /api//file:
```
а для чего во всех ручках префикс user/?
обычно используется так - user/123/file/123 т.е. после названия сущности указывается его идентификатор.

---

Да, в этот момент еще не было пользователя admin, но он теперь появился. Используется в двух ручках - создание нового эфемерного ключа и замена мастера.